libs/api/src/lib/media/novaMediaAdapter.ts

```diff
@@ -146,12 +155,35 @@ export class NovaMediaAdapter implements MediaAdapter {
     if (transformations?.sharpen) {
       queryParameters.push(`sharpen=1`)
     }
+    **/
+    let quality = transformations?.quality
+    if (quality) {
+      quality = sanitizeImageQuality(quality)
+    }
 
     // Max quality is 80 so 1 => 80
-    queryParameters.push(
-      `quality=${transformations?.quality ? Math.ceil(transformations.quality * 80) : 65}`
-    )
+    queryParameters.push(`quality=${quality ? Math.ceil(quality * 80) : 65}`)
+
+    const dto = this.parseTransformations(queryParameters)
+
+    const signature = getSignatureForImage(image.id, dto)
+    queryParameters.push(`sig=${signature}`)
 
     return encodeURI(`${this.url}/${image.id}?${queryParameters.join('&')}`)
   }
+
+  parseTransformations(params: string[]): TransformationsDto {
+    const raw: Record<string, string> = {}
+
+    for (const entry of params) {
```

Comment:
what happens in this function? This is super cryptic. Maybe a sentence or two could help to enlighten other coders =)